### PR TITLE
Fix #478 - Rename nbtrue to count_true

### LINKED ISSUE #478: [FEATURE] nbtrue, proper implementation
- [x] nbtrue(top_bits)
 - [x] use top bits implementation
 - [x] arm - proper implementation
 - [ ] If something can be done on power - do power.
 - [x]  UPD: also - can it be called `count_true`? c++ algorithm is called count.

### PR DESCRIPTION
**Pull request by jfalcou**

### CODE REVIEW

File: include/eve/module/real/algorithm/function/regular/generic/count_true.hpp [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  {
+    return v ? 1 : 0;
+  }
+
+  template<value T>
+  EVE_FORCEINLINE std::ptrdiff_t count_true_(EVE_SUPPORTS(cpu_), logical<T> v) noexcept
+  {
+    if constexpr(scalar_value<T>) return v.value() ? 1 : 0;
+    else                          return count_true(eve::detail::top_bits{v});
+  }
```
Comment (DenisYaroshevskiy) [inline include/eve/module/real/algorithm/function/regular/generic/count_true.hpp:30]: I'm not sure I like this. We don't support ignore for scalar values.
Well - I'm not sure I like scalar values used like this in general.

File: include/eve/module/real/algorithm/function/regular/generic/count_true.hpp [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+  template<simd_value T, relative_conditional_expr C>
+  EVE_FORCEINLINE std::ptrdiff_t count_true_( EVE_SUPPORTS(cpu_)
+                                            , C cond
+                                            , logical<T> v
+                                            ) noexcept
+  {
+    return eve::detail::count_true(eve::detail::top_bits{v,cond});
+  }
```
Comment (DenisYaroshevskiy) [inline include/eve/module/real/algorithm/function/regular/generic/count_true.hpp:39]: Probably the other ove should just forward to this one

File: include/eve/module/real/math/function/regular/generic/sinhc.hpp [hunk 3]
```diff
@@ -79,7 +79,7 @@ namespace eve::detail
 
         T x = abs(a0);
         auto lt1= is_less(x, one(eve::as<T>()));
-        std::size_t nb = nbtrue(lt1);
+        std::size_t nb = eve::count_true(lt1);
```
Comment (DenisYaroshevskiy) [inline include/eve/module/real/math/function/regular/generic/sinhc.hpp:82]: signed unsigned?
nb maybe should be renamed?

File: test/doc/core/count_true.cpp [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+  eve::logical<float>  xf = true;
+  eve::logical<int>    yf = false;
+
+  std::cout << "---- scalar" << '\n'
+            << "<- xf         = " << xf << '\n'
+            << "-> count_true(xf) = " << eve::count_true(xf) << '\n'
+            << "<- yf         = " << yf << '\n'
+            << "-> count_true(yf) = " << eve::count_true(yf) << '\n';
+
```
Comment (DenisYaroshevskiy) [inline test/doc/core/count_true.cpp:27]: ignore examples

File: test/unit/module/real/algorithm/count_true/regular/count_true.hpp [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+TTS_CASE_TPL("Check eve::count_true behavior with ignore", EVE_TYPE)
+{
+  eve::logical<T> data(true);
+
+  TTS_EQUAL( eve::count_true[eve::ignore_none](data)                  , EVE_CARDINAL  );
+  TTS_EQUAL( eve::count_true[eve::ignore_all](data)                   , 0             );
+  TTS_EQUAL( eve::count_true[eve::ignore_first(EVE_CARDINAL-1)](data) , 1             );
+  TTS_EQUAL( eve::count_true[eve::keep_first  (EVE_CARDINAL-1)](data) , EVE_CARDINAL-1);
+  TTS_EQUAL( eve::count_true[eve::ignore_last (EVE_CARDINAL-1)](data) , 1             );
+  TTS_EQUAL( eve::count_true[eve::keep_last   (EVE_CARDINAL-1)](data) , EVE_CARDINAL-1);
```
Comment (DenisYaroshevskiy) [inline test/unit/module/real/algorithm/count_true/regular/count_true.hpp:62]: ignore_extrema if you doing it like this?

File: include/eve/module/real/math/function/regular/generic/sinhc.hpp (hunk 3 above)
Comment (jfalcou) [reply include/eve/module/real/math/function/regular/generic/sinhc.hpp:82]: Fixed

File: test/unit/module/real/algorithm/count_true/regular/count_true.hpp (hunk 5 above)
Comment (jfalcou) [reply test/unit/module/real/algorithm/count_true/regular/count_true.hpp:62]: Fixed

File: include/eve/module/real/algorithm/function/regular/generic/count_true.hpp (hunk 1 above)
Comment (jfalcou) [reply include/eve/module/real/algorithm/function/regular/generic/count_true.hpp:30]: We have logical reduction for scalar logical for genericity purpose.
Now, ignore & logical masking doesn't make sense for scalar, hence why the conditional count_ture is simd_value constrained.

File: test/doc/core/count_true.cpp (hunk 4 above)
Comment (jfalcou) [reply test/doc/core/count_true.cpp:27]: done

**Review by DenisYaroshevskiy — APPROVED** [review]

**Pull request merged**